tests/unit/testHapiInterface.js

```diff
@@ -18,170 +18,130 @@ var lodash = require('lodash');
+          assert(errMsg instanceof ErrorMessage,
+            'The value given to sendError should be an instance of Error message');
         }
       };
-
-      var plugin = hapiInterface(fakeClient, new Configuration({serviceContext: { service: "1", version: "2" }},
-        createLogger({logLevel: 4})));
-
+      var plugin = hapiInterface(fakeClient, {
+        lacksCredentials: function () {
+          return false;
+        },
+        getVersion: function () {
+          return '1';
+        },
+        getServiceContext: function () {
+          return {service: 'node'}
+        }
+      });
       plugin.register(fakeServer, null, null, null);
-
       fakeServer.emit('request-error');
-
-      fakseServer = new EventEmitter();
-      fakeServer.ext = fakeServer.on;
-      fakeClient.sendError = function ( errMsg ) {
-
-        t.assert(
-          errMsg instanceof ErrorMessage
-          , "The value given to sendError should be an instance of Error message"
-        );
-        t.assert(
-          errMsg.serviceContext.service === testConfig._serviceContext.service
-          , "The errMsg service value and the test config service value should match"
-        );
-        t.assert(
-          errMsg.serviceContext.version === testConfig._serviceContext.version
-          , "The errMsg version value and the test config service value should match"
-        );
-        t.pass("The sendError function should be emitted when the onPreResponse event is emitted");
-      };
-
-      var testConfig = new Configuration({serviceContext: { service: "1", version: "2"  }},
-        createLogger({logLevel: 4}));
-      plugin = hapiInterface(fakeClient, testConfig);
-
-      plugin.register(fakeServer, null, function ( errMsg ) {
-
-        t.pass("Next should be called");
+    });
+  });
+  describe('Behaviour around the request/response lifecycle', function () {
+    var EVENT = 'onPreResponse';
+    var fakeServer, config, plugin, fakeClient = {sendError: function () {}};
+    before(function () {
+      config = new Configuration({
+        projectId: 'xyz',
+        serviceContext: {
+          service: 'x',
+          version: '1.x'
+        }
       });
-
-      fakeServer.emit('onPreResponse', {response: { isBoom: true }},
+      config.lacksCredentials = function () {return false;};
+      plugin = hapiInterface(fakeClient, config);
+    });
+    beforeEach(function () {
+      fakeServer = new EventEmitter();
+      fakeServer.ext = fakeServer.on;
+    });
+    afterEach(function () {
+      fakeServer.removeAllListeners();
+    });
+    it('Should call continue when a boom preResponse is emitted', function (done) {
+      plugin.register(fakeServer, null, function () {});
+      fakeServer.emit(EVENT, {response: {isBoom: true}},
         {
           continue: function () {
-
-            t.pass("The continue function should be called");
+            // The continue function should be called
+            assert(true);
```

Comment:
Also, is this the replacement for the test [here](https://github.com/GoogleCloudPlatform/cloud-errors-nodejs/pull/95#discussion_r95000037)? I don't see that test case anymore.